supplier/models.py

```diff
@@ -51,7 +54,8 @@ def delete_supplier(supplier):
         raises:
         ObjectDoesNotExists error: if the supplier is not in DB.
         """
-        Supplier.objects.get(user_name=supplier.user_name).delete()
+        # The Supplier object will be deleted automatically once the User object it references is deleted.
+        User.objects.get(username=supplier.user_name).delete()
```

Comment:
This side effect needs to be validated in a test. Same for `save_supplier`.

Reply:
They is a tests that checks that the supplier and client and for save too that they already exists after saving and they are gone after deletion same for built in user model it is guaranteed

Reply:
Can you point me to the test?